tests/test_project.py

```diff
@@ -703,6 +749,69 @@ def test_run_parallel(self):
             else:
                 self.assertFalse(job.isfile('world.txt'))
 
+    def test_run_condition_inheritance(self):
+        _project = getattr(self, 'project_class', None)
```

Comment:
I don't think this is being used.

Reply:
Never mind, this is necessary for project mocking to work below (where `self.project_class` is reassigned to). Other tests will break if we don't save this value and reassign it at the end of the function. Please add a comment to that effect.